website/blog/2024-02-08-kubernetes-frontends/index.mdx

```diff
@@ -0,0 +1,195 @@
+We are also exclusively exploring open-source and self-hosted tools and leaving the PaaS/IaaS platforms of cloud providers aside - that would be a whole different article.
+
+Setting up and interacting with your first cluster can be overwhelming. Just like me, you might have come across the infamous [`kubernetes/dashboard`](https://github.com/kubernetes/dashboard),
+followed the installation instructions, and asked yourself: "What did I just do and why exactly does this work the way it works?"
+And after some tinkering with your cluster, you might have installed even more external tools that help you with some specific aspects of cluster management,
+providing you with either a CLI or a Web UI.
+
+As a software engineer mostly engaged in web development in recent years, I was curious about how these kinds of tools are built and deployed.
+
+Let's first clarify some of the basics needed for the following exploration of different Kubernetes UIs. After that, we will see what they have in common and what makes this kind of software special, to finally form a recommendation of how one could build a Kubernetes Web UI themselves.
+
+## Kubernetes Basics
+
+The [official documentation](https://kubernetes.io/docs/concepts/overview/components/) is more than helpful anyway; there is just one important thing to remember:
+Whenever and wherever you interact with your cluster, you do it via the Kubernetes API - this holds true at least for the scope of this article,
+though there might be other use cases.
+
+As a consumer of this API, one needs to know where it is hosted and how to authenticate against it.
+The Kubernetes API can be accessed both from inside a cluster (i.e., from an application running on a pod) and outside a cluster (e.g., from your command line).
+
+Since we are looking at tools with a web UI, this UI and its backend need to be exposed such that a user can access it. The options are:
+
+- using `kubectl proxy` to open a proxy from the local machine to the cluster (see [Accessing Clusters](https://kubernetes.io/docs/tasks/access-application-cluster/access-cluster/)),
+- using `kubectl port-forward` to forward a local port to a specific pod of the cluster (see [Use Port Forwarding to Access Applications in a Cluster](https://kubernetes.io/docs/tasks/access-application-cluster/port-forward-access-application-cluster/)),
+- using a Kubernetes service of the type `LoadBalancer` to access an application of the cluster (see [Use a Service to Access an Application in a Cluster](https://kubernetes.io/docs/tasks/access-application-cluster/service-access-application-cluster/)).
+
+Alternatively, the web server could be running on the local machine of a user as well, in which case one doesn't need to worry about these options.
+Clearly, a user needs to have a valid kube config on their machine for this approach to work.
```

Comment:
I would remove the "clearly", as it would tell a reader who doesn't find it obvious that they are stupid. Maybe replace it with "however"?

Also, this is also true for at least the first two points above.